bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVM.java

```diff
@@ -1101,11 +1087,13 @@ private static Strand invokeCallable(Strand ctx, BFunctionPointer fp, FunctionCa
 
     private static int expandLongRegs(StackFrame sf, BFunctionPointer fp) {
         int longIndex = 0;
-        if (fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) > 0) {
+        if (fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) > 0 ||
+                fp.getAdditionalIndexCount(BTypes.typeByte.getTag()) > 0) {
             if (sf.longRegs == null) {
                 sf.longRegs = new long[0];
             }
-            long[] newLongRegs = new long[sf.longRegs.length + fp.getAdditionalIndexCount(BTypes.typeInt.getTag())];
+            long[] newLongRegs = new long[sf.longRegs.length + fp.getAdditionalIndexCount(BTypes.typeInt.getTag()) +
```

Comment:
Can't we simply use int index here for byte as well? Basically, fix this at the code generator level.

Reply:
Commit 4532e09 addresses this.